src/v/security/sasl_authentication.h

```diff
@@ -27,12 +31,18 @@ class sasl_mechanism {
     virtual bool failed() const = 0;
     virtual const acl_principal& principal() const = 0;
     virtual ss::future<result<bytes>> authenticate(bytes) = 0;
+    virtual std::optional<std::chrono::milliseconds>
+    credential_expires_in_ms() const {
+        return std::nullopt;
+    }
```

Comment:
@BenPope question: Does the keberos/gssapi mechanism need to implement this?  I can't remember if the ticket (or w/e it's called) we get from the client contains any timeout information

Reply:
I don't recall if there was anything useful in `lifetime_rec` in my tests. Docs: https://www.gnu.org/software/gss/reference/gss-api.html#gss-inquire-context